src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -3022,6 +2971,87 @@ export function updateSavedUserPreferenceChartIds(
     return chartSettings;
 }
 
+export async function getAllClinicalDataByStudyViewFilter(
+    studyViewFilter: StudyViewFilter
+): Promise<{ [sampleId: string]: { [attributeId: string]: string } }> {
+    const localClinicalDataCollection: ClinicalDataCollection = {
```

Comment:
i'm assuming that what's returned is ulimately just some collections of the ClinicalData type? in which case ts should be able to just infer this from the client's fetch method.

Reply:
I do not quite understand what part you are referring to. Do you object against `Promise<{ [sampleId: string]: { [attributeId: string]: string } }>` or `Promise<{ [sampleId: string]: { [attributeId: string]: string } }>`? I think using explicit types is always better than implicit? Let's discuss...